integrations/python/pymongo/workload-executor.py

```diff
@@ -35,92 +26,42 @@ def interrupt_handler(signum, frame):
-    arguments = operation_spec["arguments"]
-    for arg_name in list(arguments):
-        if arg_name == "sort":
-            sort_dict = arguments[arg_name]
-            arguments[arg_name] = list(sort_dict.items())
-    return target_name, cmd_name, arguments, operation_spec.get('result')
-
-
-def run_operation(objects, prepared_operation):
-    target_name, cmd_name, arguments, expected_result = prepared_operation
-
-    if cmd_name.lower().startswith('insert'):
-        # PyMongo's insert* methods mutate the inserted document, so we
-        # duplicate it to avoid the DuplicateKeyError.
-        arguments = copy.deepcopy(arguments)
-
-    target = objects[target_name]
-    cmd = getattr(target, cmd_name)
-    result = cmd(**dict(arguments))
-
-    if expected_result is not None:
-        if isinstance(result, Cursor) or isinstance(result, CommandCursor):
-            result = list(result)
-        assert result == expected_result
-
-
-def connect(mongodb_uri):
-    if WIN32 and mongodb_uri.startswith("mongodb+srv://"):
-        # TODO: remove this once BUILD-10841 is done.
-        import certifi
-        return MongoClient(mongodb_uri, tlsCAFile=certifi.where())
-    return MongoClient(mongodb_uri)
-
-
 def workload_runner(mongodb_uri, test_workload):
-    # Do not modify connection string and do not add any extra options.
-    client = connect(mongodb_uri)
-
-    # Create test entities.
-    database = client.get_database(test_workload["database"])
-    collection = database.get_collection(test_workload["collection"])
-    objects = {"database": database, "collection": collection}
-
-    # Run operations
-    num_failures = 0
-    num_errors = 0
-    num_operations = 0
-    global IS_INTERRUPTED
-
-    operations = test_workload["operations"]
-    ops = [prepare_operation(op) for op in operations]
-
-    while True:
-        if IS_INTERRUPTED:
-            break
-        for op in ops:
-            try:
-                run_operation(objects, op)
-            except AssertionError:
-                traceback.print_exc()
-                num_failures += 1
-            except Exception:
-                traceback.print_exc()
-                num_errors += 1
-            else:
-                num_operations += 1
-
-    # We reach here once IS_INTERRUPTED has been set to True.
-    stats = {"numErrors": num_errors, "numFailures": num_failures,
-             "numSuccesses": num_operations}
-    print("Workload statistics: {!r}".format(stats))
-
-    sentinel = os.path.join(os.path.abspath(os.curdir), 'results.json')
-    print("Writing statistics to sentinel file {!r}".format(sentinel))
-    with open('results.json', 'w') as fr:
-        json.dump(stats, fr)
-    exit(0 or num_errors or num_failures)
+    runner = UnifiedSpecTestMixinV1()
+    runner.TEST_SPEC = test_workload
+    UnifiedSpecTestMixinV1.TEST_SPEC = test_workload
+    runner.setUp()
+    runner.client = pymongo.MongoClient(mongodb_uri)
```

Comment:
This is necessary because there does not seem to exist a localhost:27017 instance of mongodb, so this instead overrides the client with the mongodb_uri that is where mongodb is actually located.

Reply:
Please explain this in a comment as well.

Reply:
Done.